File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        vector<int> mp(26, 0);
        for(char c : chars) mp[c-'a']++;
        int ans = 0;
        for(string s : words){
            vector<int> v(26, 0);
            for(char c : s) v[c-'a']++;
            bool isValid = true;
            for(char c : s){
                if(mp[c-'a'] < v[c-'a']){
                    isValid = false;
                    break;
                }
            }
            ans += isValid ? s.size() : 0;
        }
        return ans;
    }
};

File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        mp = [0] * 26
        for c in chars: mp[ord(c)-ord('a')]+=1
        ans = 0
        for s in words:
            cnt = [0] * 26
            for c in s: cnt[ord(c)-ord('a')] += 1
            isValid = True
            for c in s:
                if mp[ord(c)-ord('a')] < cnt[ord(c)-ord('a')]:
                    isValid = False
                    break
            if isValid:
                ans += len(s)
        return ans